client-tags/typing.md

```diff
@@ -27,14 +27,14 @@ For the purpose of this specification, a message is defined as a user-generated
 ## Format
 The `typing` tag is sent as a [`TAGMSG`][tags]. The client tag takes no parameters.
 
-The `typing` client tag SHOULD be sent by the client when the first character is typed in the text-input field. If new text is entered into the text-input field following sending of the text-input field content, the client MUST send a new `typing` tag after sending the preceding content.
+The `typing` client tag SHOULD be sent by the client when the first character is typed in the text-input field and after every 3 seconds of a non-empty text-input field to avoid a state change. If new text is entered into the text-input field following sending of the text-input field content, the client MUST send a new `typing` tag after sending the preceding content.
```

Comment:
How about replacing this paragraph with something like:

> The typing notification SHOULD be sent by clients whenever any update is made to the text-input field. Input event handlers SHOULD be throttled so that typing notifications are not sent within 3 seconds of each other.

This allows clients to throttle more aggressively if they choose, but I think that's OK.